### PR TITLE
feat: Add support for handling refs within OpenAPI path items

### PR DESCRIPTION
**Pull request by DerDackel**
* **Please check if the PR fulfills these requirements**
- [x] The commit message describes your change
- [x] Tests for the changes have been added if possible (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Changes are mentioned in the changelog (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Fixes a bug due to a recently addressed gap in the OpenAPI spec implementation of openapi-typescript where refs to path items within the `paths` section of the API defnition weren't supported. This resulted in a type signature change on openapi-typescripts `PathsObject` that broke the RestApi projen aspect. 

I took the liberty of adding some error handling for processing malformed OpenAPI definitions (read below), so some definitions where older versions of cdk-serverless would have ignored e.g. invalid refs will now throw an error upon project synthesiziation.

* **What is the current behavior?** (You can also link to an open issue here)

Project won't compile with recent versions of openapi-typescript due to a type signature change in the dependency.
Current behavior would alreadys have guarded against accidentally creating superfluous lambdas for refs otherwise as the RestApi component only follows paths in the definition where the child has keys that correspond to the keys of a `PathItemObject`.

Refs introduce two error states though: Cyclical refs and refs to nonexistant or malformed nodes in the definition. Currently openapi-typescript will terminate on cyclical references to avoid endless loops, but will gladly generate types that reference paths that don't exist, or that are not path items at all. This might probably occur in other parts of the code generation too.

While someone can probably come up with a case that can hack this into some nifty feature, I couldn't and would personally prefer it the synthesizer stops and delivers a specific error rather than wondering what the type checker is complaining about. So I have the RestApi component check whether any path item refs in the definition file point to valid path items. Cycle checks are a required necessity for this.

If this behavior is explicitly not desired, please close this and I'll submit a minimal fix to satisfy the compiler as a new PR.

* **What is the new behavior (if this is a feature change)?**

cdk-serverless will compile again with newer versions of openapi-typescript. Runtime behavior will now throw errors upon encountering OpenAPI definitions with invalid or cyclical path item references.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their setup due to this PR?)

No API changes were introduced, but abovementioned changes may affect existing projects making use of existing unspecified/error-tolerant behavior.

* **Other information**:

This will hopefully be enough to make cdk-serverless build again with updated dependencies

### CODE REVIEW

File: src/projen/rest-api.ts [hunk 1]
```diff
@@ -43,10 +48,10 @@ export class RestApi extends pj.Component {
         }
       }
     }
-    if (!fs.existsSync('./src/generated')) {
-      fs.mkdirSync('./src/generated');
+    if (!fs.existsSync(`${this.project.outdir}/src/generated`)) {
```
Comment (hoegertn) [inline src/projen/rest-api.ts:51]: Love it

Comment (DerDackel) [reply src/projen/rest-api.ts:51]: Oh, completely forgot to mention this! I stumbled over these hardcoded paths when writing unit tests. I hope this doesn't break something else!

Comment (hoegertn) [reply src/projen/rest-api.ts:51]: Will check

**Comment by hoegertn** [comment]
Thanks I will review it and merge it afterwards

File: src/projen/rest-api.ts (hunk 1 above)
Comment (DerDackel) [reply src/projen/rest-api.ts:51]: Datastore and GraphQL aspects share the same relative paths, my gut says if it breaks something, I'd rather fix that than leave the old paths, even if currently unit testing was the first time I ran into problems with this.

**Comment by DerDackel** [comment]
Oh, another thing I forgot: The `synthSnapshotForAspect` function I wrote for testing utility as based on a similar utility function within projen, as I basically looked up how they test project synthesis. I basically took their `synthSnapshot` function and amended it so I could do a two-step synthesizing a base project and synthesizing again after applying some side-effects to it that require an existing synthesized state.

**Comment by hoegertn** [comment]
@DerDackel please rebase onto merged PR #277

**Comment by DerDackel** [comment]
Done!

**Review by hoegertn — APPROVED** [review]

**Pull request merged**